docs/reference/troubleshooting/troubleshooting-searches.asciidoc

```diff
@@ -0,0 +1,240 @@
+
+[[troubleshooting-searches]]
+== Troubleshooting searches
+
+When you query your data, Elasticsearch may return an error, no search results,
+or results in an unexpected order. This guide describes how to troubleshoot
+searches.
+
+[discrete]
+[[troubleshooting-searches-exists]]
+=== Ensure the data stream, index, or alias exists
+
+Elasticsearch returns an `index_not_found_exception` when the data stream, index
+or alias you try to query does not exist. This can happen when you misspell the
+name or when the data has been indexed to a different data stream or index.
+
+Use the <<indices-exists,exists API>> to check whether a data stream, index, or
+alias exists:
+
+[source,console]
+----
+HEAD my-data-stream
+----
+
+Use the <<data-stream-stats-api,data stream stats API>> to list all data
+streams:
+
+[source,console]
+----
+GET /_data_stream/_stats?human=true
+----
+
+Use the <<cat-indices,cat indices API>> to list all indices:
+
+[source,console]
+----
+GET _cat/indices?v=true&s=index&h=index,docs.count,status,health
+----
+
+Use the <<cat-alias,cat aliases API>> to list all aliases:
```

Comment:
having to check two cat API, one for indices and one for aliases is not great from a users perspective. Should we rather point to an API that returns each index together with the aliases they are associated with?

Reply:
That's a good idea. How about `GET _all?filter_path=*.aliases`?

Reply:
can you post the output? :)

Reply:
It outputs a list of indices, with the aliases for each index:

```
{
  ".ds-logs-enterprise_search.api-default-2022.08.29-000001": {
    "aliases": {}
  },
  ".ds-logs-enterprise_search.audit-default-2022.08.29-000001": {
    "aliases": {}
  },
  "image-index": {
    "aliases": {}
  },
  "kibana_sample_data_ecommerce": {
    "aliases": {}
  },
  "kibana_sample_data_flights": {
    "aliases": {}
  },
  "kibana_sample_data_logs": {
    "aliases": {}
  },
  "my-index-000001": {
    "aliases": {
      "my-alias": {}
    }
  },
  "product-index": {
    "aliases": {}
  }
}
```

Reply:
looks neat, thanks!